crossplane/ext/lua.py

```diff
@@ -6,6 +6,25 @@
 from crossplane.ext.abstract import CrossplaneExtension
 
 
+class EmplaceIter:
+    def __init__(self, it):
+        self.it = it
+        self.ret = []
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        if len(self.ret) > 0:
+            v = self.ret.pop()
+            return v
+        return next(self.it)
+
```

Comment:
To add Python 2 support, adding something like this should work:
```suggestion

    next = __next__

```
(https://portingguide.readthedocs.io/en/latest/iterators.html#new-iteration-protocol-next)